File: backend/routers/resumen.py
from fastapi import APIRouter, Query
import sqlite3
import pandas as pd
from typing import List, Optional

router = APIRouter(prefix="/api/resumen")
DB_PATH = "/Users/danielramirezquintana/Desktop/TOK3M_1/DB/TOKEM.db"

@router.get("/graficos")
def get_graficos(inicio: Optional[str]=None, fin: Optional[str]=None, codigos: Optional[List[str]]=Query(None), empresas: Optional[List[str]]=Query(None), ejecutivos: Optional[List[str]]=Query(None)):
    conn = sqlite3.connect(DB_PATH)
    f = []
    if inicio and fin:
        f.append(f"YMD BETWEEN '{inicio.replace('-', '')}' AND '{fin.replace('-', '')}'")
    if codigos:
        vals = ", ".join([f"'{c}'" for c in codigos])
        f.append(f"CODIGO_CONTACTO IN ({vals})")
    if empresas:
        vals = ", ".join([f"'{e}'" for e in empresas])
        f.append(f"EMPRESA IN ({vals})")
    if ejecutivos:
        vals = ", ".join([f"'{j}'" for j in ejecutivos])
        f.append(f"NOMBRE_EJECUTIVO IN ({vals})")
    
    where = "WHERE " + " AND ".join(f) if f else ""
    
    df_dia = pd.read_sql(f"SELECT YMD as FECHA, SUM(cantidad) as cantidad FROM ANALISIS_TOK3M {where} GROUP BY YMD ORDER BY YMD", conn)
    df_emp = pd.read_sql(f"SELECT EMPRESA, SUM(cantidad) as cantidad FROM ANALISIS_TOK3M {where} GROUP BY EMPRESA ORDER BY cantidad DESC", conn)
    df_con = pd.read_sql(f"SELECT CODIGO_CONTACTO, SUM(cantidad) as cantidad FROM ANALISIS_TOK3M {where} GROUP BY CODIGO_CONTACTO ORDER BY cantidad DESC", conn)
    df_eje = pd.read_sql(f"SELECT NOMBRE_EJECUTIVO, SUM(cantidad) as cantidad FROM ANALISIS_TOK3M {where} GROUP BY NOMBRE_EJECUTIVO ORDER BY cantidad DESC", conn)
    conn.close()
    
    return {
        "por_dia": df_dia.to_dict(orient='records'),
        "por_empresa": df_emp.to_dict(orient='records'),
        "por_contacto": df_con.to_dict(orient='records'),
        "por_ejecutivo": df_eje.to_dict(orient='records')
    }

File: server/routes/resumen.js
const express = require('express');
const router = express.Router();

router.get('/graficos', async (req, res) => {
    const pool = req.app.get('pool');
    const { inicio, fin, empresas, ejecutivos, contactos } = req.query;
    let filters = [];

    // Filtros ajustados a la nueva estructura
    if (inicio && fin) filters.push(`ymd BETWEEN ${inicio.replace(/-/g, '')} AND ${fin.replace(/-/g, '')}`);
    if (empresas) filters.push(`empresa = ANY(string_to_array('${empresas}', ','))`);
    
    // IMPORTANTE: nombre_ejecutivo y codigo_contacto son los nombres en la tabla maestro
    if (ejecutivos) filters.push(`nombre_ejecutivo = ANY(string_to_array('${ejecutivos}', ','))`);
    if (contactos) filters.push(`codigo_contacto = ANY(string_to_array('${contactos}', ','))`);

    const where = filters.length > 0 ? `WHERE ${filters.join(' AND ')}` : '';

    try {
        // Ahora todas las consultas atacan a resumen_maestro
        const [dia, eje, emp, con] = await Promise.all([
            pool.query(`SELECT ymd as "FECHA", SUM(total_gestiones) as "cantidad" FROM resumen_maestro ${where} GROUP BY ymd ORDER BY ymd`),
            pool.query(`SELECT nombre_ejecutivo as "NOMBRE_EJECUTIVO", SUM(total_gestiones) as "cantidad" FROM resumen_maestro ${where} GROUP BY nombre_ejecutivo`),
            pool.query(`SELECT empresa as "EMPRESA", SUM(total_gestiones) as "cantidad" FROM resumen_maestro ${where} GROUP BY empresa`),
            pool.query(`SELECT codigo_contacto as "CODIGO_CONTACTO", SUM(total_gestiones) as "cantidad" FROM resumen_maestro ${where} GROUP BY codigo_contacto`)
        ]);
        
        res.json({ 
            por_dia: dia.rows, 
            por_ejecutivo: eje.rows, 
            por_empresa: emp.rows, 
            por_contacto: con.rows 
        });
    } catch (e) { 
        console.error("Error en router resumen:", e);
        res.status(500).json({ error: e.message }); 
    }
});

module.exports = router;